Header implementation

Costaknc opened the pull request:
Έφτιαξα το menu, το burger menu και να ειναι λειτουργικο (να ανοιγει/κλεινει δηλαδη ενα menu απο κατω του).

Costaknc committed:
first navbar implementation

konpetrogian committed:
responsivness+'burger menu' implementation

kondritsas commented on src/app/playground/playground.component.html:
```diff
@@ -1 +1,60 @@
+    class="aa-grow aa--items-left aa--mr-[5vw] aa--flex aa--py-2 aa--text-2xl md:aa--hidden"
+    (click)="onClick()"
+  >
+    <span class="material-symbols-outlined aa--cursor-pointer hover:aa--text-orange-500">
```
You can use instead the font awesome library which is already installed in the package.json.  Import FontAwesomeModule and use faBars icon from @fortawesome/free-solid-svg-icons. To render the icon in the template use <fa-icon> element

Example: <fa-icon class='icon aa--text-[#f6efef] aa--h-[30px] aa--mt-[3px]'[icon]="faBriefcase" size="2x" />

kondritsas reviewed (changes requested):
Check for icon change to use the already installed packages

iliasanthimos commented on src/app/playground/playground.component.ts:
```diff
@@ -7,5 +7,18 @@ import { Component } from '@angular/core';
 })
 export class PlaygroundComponent {
-
+  classContent =
```
You can simplify it like.
```
  isMenuOpen = false;

this is to handle open - close in case of resizing
@HostListener('window:resize', ['$event'])
  onResize(event: Event): void {
    const windowWidth = (event.target as Window).innerWidth;
    if (windowWidth >= 768 && this.isMenuOpen) {
      this.isMenuOpen = false;
    }
  }

  toggleMenu(): void {
    this.isMenuOpen = !this.isMenuOpen;
  }
```

```
<button
      class="aa--block aa--text-white focus:aa--outline-none md:aa--hidden"
      (click)="toggleMenu()"
      aria-label="Toggle Menu"
    >
      <fa-icon [icon]="isMenuOpen ? faTimes : faBars" />
    </button>
```

to group the items you can use something like 

```
<ng-template #menuItems>
    <a routerLink="/" class="aa--block hover:aa--underline">Jobs</a>
    <a routerLink="/favorites" class="aa--block hover:aa--underline">Favorites</a>
    <button (click)="toggleTheme()">
      <fa-icon [icon]="themeService.isDarkMode() ? faSun : faMoon" />
    </button>
  </ng-template>
```

and render them in the correct place

iliasanthimos commented on src/app/playground/playground.component.html:
```diff
@@ -1 +1,60 @@
-<p>playground works!</p>
+<header
```
We can create a global header component to include the following code

iliasanthimos commented on src/index.html:
```diff
@@ -5,7 +5,11 @@
     <base href="/" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
-    <link rel="icon" type="image/x-icon"/>
+    <link rel="icon" type="image/x-icon" />
```
based on @kondritsas comment about icons you can remove this

Costaknc committed:
created-different0components

created the different navbar component and added a dummy app logo just for context

Costaknc committed:
fixed_mobile_menu

Fixed some icons. Fixed the menu to be created with ngFor/ngIf etc

Costaknc committed:
removed playground, added header component

Costaknc commented:
Finished all the changes requested above by all. We can review the code again.

Costaknc committed:
header/navbar margin fine-tuning

iliasanthimos committed:
Merge branch 'development' into header-implementation

# Conflicts:
#	src/app/app.component.ts
#	src/app/playground/playground.component.html
#	src/app/playground/playground.component.ts
